src/main/java/practice/StreamPractice.java

```diff
@@ -75,6 +111,11 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * parametrized with Candidate in CandidateValidator.
      */
     public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+        CandidateValidator candidateValidator = new CandidateValidator();
```

Comment:
```suggestion
        Predicate<Candidate> candidateValidator = new CandidateValidator();
```
please use abstraction for variable type